### PR TITLE
When manualNamespaceNames is enabled, attach labels the same was as done for pre-created-ns

### PR DESCRIPTION
**Pull request by rujhan-arora-astronomer**
## Description

When manualNamespaceNames is enabled, attach labels the same was as done for pre-created-ns
## Related Issues

https://github.com/astronomer/issues/issues/4650

## Testing

Locally

## Merging

0.32.1
0.30.7

### CODE REVIEW

**Comment by pgvishnuram** [comment]
@Priyanka13Astro - we have covered all namespace related configs.

**Comment by rujhan-arora-astronomer** [comment]
@pgvishnuram, Plz review and merge it too.

File: charts/prometheus/templates/prometheus-config-configmap.yaml [hunk 1]
```diff
@@ -270,7 +270,7 @@ data:
           - regex: 'label_kubernetes_pod_operator'
             action: labeldrop
           # Required for multi-namespace mode
-          {{- if or .Values.global.features.namespacePools.enabled .Values.global.namespaceFreeFormEntry }}
+          {{- if or .Values.global.features.namespacePools.enabled .Values.global.namespaceFreeFormEntry .Values.global.manualNamespaceNamesEnabled}}
```
Comment (danielhoherd) [inline charts/prometheus/templates/prometheus-config-configmap.yaml:273]: Seems like we should update the comment above this line.

File: charts/prometheus/templates/prometheus-config-configmap.yaml [hunk 2]
```diff
@@ -269,8 +269,8 @@ data:
             action: labeldrop
           - regex: 'label_kubernetes_pod_operator'
             action: labeldrop
-          # Required for multi-namespace mode
-          {{- if or .Values.global.features.namespacePools.enabled .Values.global.namespaceFreeFormEntry }}
+          # Required for multi-namespace mode (non auto-generated namespaces should be relabelled differently)
```
Comment (danielhoherd) [inline charts/prometheus/templates/prometheus-config-configmap.yaml:272]:
```suggestion
          # Required for multi-namespace mode (non auto-generated namespaces should be relabeled differently)
```

**Review by danielhoherd — APPROVED** [review]
```
____/\\\_________________/\\\\\\\\\\\\__/\\\\\\\\\\\\\\\__/\\\\____________/\\\\___________
____\/\\\_______________/\\\//////////__\///////\\\/////__\/\\\\\\________/\\\\\\__________
_____\/\\\______________/\\\___________________\/\\\_______\/\\\//\\\____/\\\//\\\_________
______\/\\\_____________\/\\\____/\\\\\\\_______\/\\\_______\/\\\\///\\\/\\\/_\/\\\________
_______\/\\\_____________\/\\\___\/////\\\_______\/\\\_______\/\\\__\///\\\/___\/\\\_______
________\/\\\_____________\/\\\_______\/\\\_______\/\\\_______\/\\\____\///_____\/\\\______
_________\/\\\_____________\/\\\_______\/\\\_______\/\\\_______\/\\\_____________\/\\\_____
__________\/\\\\\\\\\\\\\\\_\//\\\\\\\\\\\\/________\/\\\_______\/\\\_____________\/\\\____
___________\///////////////___\////////////__________\///________\///______________\///____
```

File: tests/chart_tests/test_prometheus_config_configmap.py [hunk 3]
```diff
@@ -311,3 +311,44 @@ def test_prometheus_config_release_relabel_with_pre_created_namespace(
...
+        assert metric_relabel_config_search_result[0]["source_labels"] == ["pod"]
+        assert metric_relabel_config_search_result[0]["replacement"] == "$1"
+        assert metric_relabel_config_search_result[0]["target_label"] == "release"
+
+        assert metric_relabel_config_search_result[1]["regex"] == "(.+)-resource-quota$"
+        assert metric_relabel_config_search_result[1]["source_labels"] == [
+            "resourcequota"
+        ]
+        assert metric_relabel_config_search_result[1]["replacement"] == "$1"
+        assert metric_relabel_config_search_result[1]["target_label"] == "release"
```
[the remark this answers is not in the document]
Comment (danielhoherd) [reply tests/chart_tests/test_prometheus_config_configmap.py]: Solved by pulling assertions out into a new function.

**Comment by rujhan-arora-astronomer** [comment]
Hi @danielhoherd  Plz merge it too, I don't have the permissions

**Pull request merged**